viewer/Main.cpp

```diff
@@ -19,15 +20,46 @@ int main(int argc, char * argv[])
   qmlRegisterType<ImageCollection>("Hdrv", 1, 0, "ImageCollection");
   qmlRegisterType<ImageArea>("Hdrv", 1, 0, "ImageArea");
   qmlRegisterType<Settings>("Hdrv", 1, 0, "Settings");
+  qmlRegisterType<IPCServer>("Hdrv", 1, 0, "Server");
+  qmlRegisterType<IPCClient>("Hdrv", 1, 0, "Client");
 
   Settings settings;
+  IPCServer server;
+  IPCClient client;
   ImageCollection images;
   QQmlApplicationEngine engine;
   engine.rootContext()->setContextProperty("settings", &settings);
   engine.rootContext()->setContextProperty("images", &images);
+  engine.rootContext()->setContextProperty("server", &server);
+  engine.rootContext()->setContextProperty("client", &client);
 
+  QObject::connect(&server, SIGNAL(openFile(QUrl const &)), &images, SLOT(load(QUrl const &)));
+  QObject::connect(&server, &IPCServer::openFile, [] (QUrl const &) {
```

Comment:
You want to have the code in a separate function 'moveToForeground', instead of a lambda function? Ok,